Sources/Commands/SwiftPackageTool.swift

```diff
@@ -863,6 +865,138 @@ extension SwiftPackageTool {
+    // recognized as a built-in command will cause `swift-package` to search for plugin commands, instead of using
+    // a separate `plugin` subcommand for this.
+    struct PluginCommand: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            commandName: "plugin",
+            abstract: "Invoke a command plugin (note: the use of a `plugin` subcommand for this is temporary)"
+        )
+
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+
+        /// The specific target to apply the plugin to.
+        @Option(name: .customLong("target"), help: "Target(s) to which the plugin command should be applied")
+        var targetNames: [String] = []
+
+        @Argument(help: "Name of the command plugin to invoke")
+        var command: String
+
+        @Argument(help: "Any arguments to pass to the plugin")
+        var arguments: [String] = []
+        
+        func findPlugins(matching command: String, in graph: PackageGraph) -> [PluginTarget] {
+            // Find all the available plugin targets.
+            let availablePlugins = graph.allTargets.compactMap{ $0.underlyingTarget as? PluginTarget }
+            
+            // Find and return the plugins that match the command.
+            return availablePlugins.filter {
+                // Filter out any non-command plugins.
+                guard case .command(let intent, _) = $0.capability else { return false }
+                // FIXME: We shouldn't hardcode these verbs.
+                switch intent {
+                case .documentationGeneration:
+                    return command == "generate-documentation"
+                case .sourceCodeFormatting:
+                    return command == "format-source-code"
+                case .custom(let verb, _):
+                    return command == verb
+                }
+            }
+        }
+
+        func run(_ swiftTool: SwiftTool) throws {
+            // Load the workspace and resolve the package graph.
+            let packageGraph = try swiftTool.loadPackageGraph()
+            
+            // Find the plugins that match the command.
+            let matchingPlugins = findPlugins(matching: command, in: packageGraph)
+
+            // Complain if we didn't find exactly one.
+            if matchingPlugins.isEmpty {
+                swiftTool.observabilityScope.emit(error: "No plugins found for '\(command)'")
+                throw ExitCode.failure
+            }
+            else if matchingPlugins.count > 1 {
+                swiftTool.observabilityScope.emit(error: "\(matchingPlugins.count) plugins found for '\(command)'")
+                throw ExitCode.failure
+            }
+            
+            // At this point we know we found exactly one command plugin, so we run it.
+            let plugin = matchingPlugins[0]
+            print("Running plugin \(plugin)")
```

Comment:
This line is temporary.  Most likely we'll want a `verbose` flag to control whether and how to emit status output.

Reply:
use `swiftTool.observabilityScope.emit(info: `?

Reply:
Great thanks — will change to that.